### PR TITLE
CBG-241 - Remove duplicate sgcollect_info files

### PR DESCRIPTION
**Pull request by JFlath**

### CODE REVIEW

File: rest/api_test.go [hunk 1]
```diff
@@ -68,6 +68,31 @@ func TestRoot(t *testing.T) {
...
+	var body db.Body
+	json.Unmarshal(response.Body.Bytes(), &body)
+	goassert.Equals(t, body["db_name"], "db")
+	goassert.Equals(t, body["state"], "Online")
+	goassert.Equals(t, body["db_name"] != nil,true)
+	goassert.Equals(t, body["update_seq"] != nil,true)
+	goassert.Equals(t, body["committed_update_seq"] != nil,true)
+	goassert.Equals(t, body["compact_running"] != nil,true)
+	goassert.Equals(t, body["purge_seq"] != nil,true)
+	goassert.Equals(t, body["disk_format_version"] != nil,true)
```
Comment (JFlath) [inline rest/api_test.go]: Sure there's a fancy way of doing this with an unmarshal into a struct...

File: rest/api.go [hunk 2]
```diff
@@ -182,39 +182,56 @@ func (h *handler) handlePostUpgrade() error {
 	return nil
 }
 
-func (h *handler) instanceStartTime() json.Number {
-	return json.Number(strconv.FormatInt(h.db.StartTime.UnixNano()/1000, 10))
+func instanceStartTime(dbc *db.DatabaseContext) json.Number {
```
Comment (JFlath) [inline rest/api.go]: Refactored this to from a method to a function as if we're using this from `/_status` we can't rely on `h.db` and need to pass in the `DatabaseContext` explicitly.

File: rest/utilities_testing.go [hunk 3]
```diff
@@ -1432,6 +1434,13 @@ type TestLogger struct {
 	T *testing.T
 }
 
+func assertBodySatisfiesModel(t *testing.T, got db.Body, model db.Body) {
```
Comment (JFlath) [inline rest/utilities_testing.go]: Should almost certainly be in goassert, but...

**Comment by JFlath** [comment]
@JRascagneres As discussed, this should now be what's expected in this PR

**Comment by JRascagneres** [comment]
Looks good to me. Seems to fulfill the requirements of the jira ticket

File: tools/sgcollect_info [hunk 4]
```diff
@@ -607,8 +564,12 @@ def make_sg_tasks(zip_dir, sg_url, sync_gateway_config_path_option, sync_gateway
...
     config_tasks = make_config_tasks(zip_dir, sg_config_path, sg_url, should_redact)
 
-    # Curl the / endpoint and /db endpoints and save output
-    status_tasks = make_status_tasks(sg_url, should_redact)
+    # Curl the /_status
+    status_tasks = make_curl_task(name="Collect server status",
+                                  user="",
+                                  password="",
+                                  url="{0}/_status".format(sg_url),
+                                  log_file="sync_gateway.log")
```
Comment (bbrks) [inline tools/sgcollect_info]: If multiple tasks are assigned to the same log_file, will they be appended, or overwritten? (looks like Running server config above got renamed to match this one)

Comment (JRascagneres) [reply tools/sgcollect_info]: They get appended to the file.

File: rest/admin_api.go [hunk 5]
```diff
@@ -340,6 +340,35 @@ func (h *handler) handleGetLogging() error {
 	return nil
 }
 
+func (h *handler) handleGetStatus() error {
+	status := db.Body{
+		"databases":    make(map[string]db.Body),
+		"active_tasks": h.server.replicator.ActiveTasks(),
+		"version":      base.LongVersionString,
+		"vendor":       db.Body{"name": base.ProductName, "version": base.VersionNumber},
+	}
```
Comment (adamcfraser) [inline rest/admin_api.go:371]: Add a struct for status instead of building the Body piece-wise.

File: rest/api_test.go [hunk 6]
```diff
@@ -68,6 +69,38 @@ func TestRoot(t *testing.T) {
...
+		"db_name":              reflect.String,
+		"state":                reflect.String,
+		"update_seq":           reflect.Float64,
+		"committed_update_seq": reflect.Float64,
+		"purge_seq":            reflect.Float64,
+		"disk_format_version":  reflect.Float64,
+		"compact_running":      reflect.Bool,
+	}
+
+	assertBodySatisfiesModel(t, body, model)
```
Comment (adamcfraser) [inline rest/api_test.go]: Can remove this assert by using the struct for status.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

**Comment by JFlath** [comment]
@JRascagneres Mind holding off on this for a sec, IIRC there was a specific reason for the design used here

File: rest/api.go [hunk 7]
```diff
@@ -199,19 +211,19 @@ func (h *handler) handleGetDB() error {
...
-		"purge_seq":            0, // TODO: Should track this value
-		"disk_format_version":  0, // Probably meaningless, but add for compatibility
-		"state":                runState,
+	var response = DatabaseRoot{
+		DBName:                        h.db.Name,
+		SequenceNumber:                lastSeq,
+		CommittedUpdateSequenceNumber: lastSeq,
+		InstanceStartName:             h.instanceStartTime(),
+		CompactRunning:                h.db.IsCompactRunning(),
+		PurgeSequenceNumber:           0,
```
Comment (bbrks) [inline rest/api.go]: Maintain old comments

```suggestion
		PurgeSequenceNumber:           0, // TODO: Should track this value
```

File: rest/api.go [hunk 8]
```diff
@@ -199,19 +211,19 @@ func (h *handler) handleGetDB() error {
...
+		DiskFormatVersion:             0,
```
Comment (bbrks) [inline rest/api.go]:
```suggestion
		DiskFormatVersion:             0, // Probably meaningless, but add for compatibility
```

File: rest/admin_api_test.go [hunk 9]
```diff
@@ -565,6 +565,26 @@ func TestLoggingCombined(t *testing.T) {
...
+	rt := NewRestTester(t, &rtConfig)
+	defer rt.Close()
+
+	response := rt.SendRequest("GET", "/_status", "")
+	assertStatus(t, response, 404)
+
+	response = rt.SendAdminRequest("GET", "/_status", "")
+	assertStatus(t, response, 200)
+	var responseBody Status
+	json.Unmarshal(response.Body.Bytes(), &responseBody)
```
Comment (bbrks) [inline rest/admin_api_test.go]: We're not checking this response is unmarshalled into the struct successfully (missing error check)

File: rest/api.go [hunk 10]
```diff
@@ -186,6 +186,18 @@ func (h *handler) instanceStartTime() json.Number {
 	return json.Number(strconv.FormatInt(h.db.StartTime.UnixNano()/1000, 10))
 }
 
+type DatabaseRoot struct {
+	DBName                        string      `json:"db_name"`
+	SequenceNumber                uint64      `json:"update_seq"`
+	CommittedUpdateSequenceNumber uint64      `json:"comitted_update_seq"`
+	InstanceStartName             json.Number `json:"instance_start_name"`
```
Comment (bbrks) [inline rest/api.go]: Name? Are you sure?

File: rest/api.go [hunk 11]
```diff
@@ -186,6 +186,18 @@ func (h *handler) instanceStartTime() json.Number {
...
+	CompactRunning                bool        `json:"compact_running"`
+	PurgeSequenceNumber           uint64      `json:"purge_seq"`
+	DiskFormatVersion             uint64      `json:"disk_format_version"`
+	State                         string      `json:"state"`
+	ServerUUID                    string      `json:"server_uuid"`
```
Comment (bbrks) [inline rest/api.go]: Let's keep the API behaviour the same by specifying omitempty for this field.

```suggestion
	ServerUUID                    string      `json:"server_uuid,omitempty"`
```

File: rest/api.go [hunk 12]
```diff
@@ -199,19 +211,19 @@ func (h *handler) handleGetDB() error {
...
+		InstanceStartName:             h.instanceStartTime(),
+		CompactRunning:                h.db.IsCompactRunning(),
+		PurgeSequenceNumber:           0,
+		DiskFormatVersion:             0,
+		State:                         runState,
 	}
 
 	if uuid := h.db.DatabaseContext.GetServerUUID(); uuid != "" {
-		response["server_uuid"] = uuid
+		response.ServerUUID = uuid
```
Comment (bbrks) [inline rest/api.go]: Move up into struct initialisation after omitempty is added

File: rest/api_test.go [hunk 13]
```diff
@@ -68,6 +68,26 @@ func TestRoot(t *testing.T) {
...
 
+func TestDBRoot(t *testing.T) {
+	rtConfig := RestTesterConfig{NoFlush: true}
+	rt := NewRestTester(t, &rtConfig)
+	defer rt.Close()
+
+	response := rt.SendRequest("GET", "/db/", "")
+	assertStatus(t, response, 200)
+	var body db.Body
+	json.Unmarshal(response.Body.Bytes(), &body)
```
Comment (bbrks) [inline rest/api_test.go]: missing error check again

**Review by bbrks — CHANGES_REQUESTED** [review]
We've not maintained the same API here - and the updated tests don't quite test the struct either.

**Comment by bbrks** [comment]
> @JRascagneres Mind holding off on this for a sec, IIRC there was a specific reason for the design used here

The `assertBodySatisfiesModel` thing? Or something else in this PR?

File: rest/api.go [hunk 14]
```diff
@@ -186,6 +186,18 @@ func (h *handler) instanceStartTime() json.Number {
 	return json.Number(strconv.FormatInt(h.db.StartTime.UnixNano()/1000, 10))
 }
 
+type DatabaseRoot struct {
+	DBName                        string      `json:"db_name"`
+	SequenceNumber                uint64      `json:"update_seq"`
+	CommittedUpdateSequenceNumber uint64      `json:"comitted_update_seq"`
+	InstanceStartTime             json.Number `json:"instance_start_name"`
```
Comment (bbrks) [inline rest/api.go]: Still have `instance_start_name` here.

File: rest/api.go [hunk 15]
```diff
@@ -186,6 +186,18 @@ func (h *handler) instanceStartTime() json.Number {
 	return json.Number(strconv.FormatInt(h.db.StartTime.UnixNano()/1000, 10))
 }
 
+type DatabaseRoot struct {
+	DBName                        string      `json:"db_name"`
+	SequenceNumber                uint64      `json:"update_seq"`
+	CommittedUpdateSequenceNumber uint64      `json:"comitted_update_seq"`
```
Comment (bbrks) [inline rest/api.go]: Typo'd committed?

**Review by bbrks — CHANGES_REQUESTED** [review]
Still have some incorrect field names in the `DatabaseRoot` struct

**Comment by adamcfraser** [comment]
@JFlath Going to go ahead and merge as-is based on our review, but open to a followup discussion as needed for any design concerns.

**Review by adamcfraser — APPROVED** [review]

**Review by bbrks — APPROVED** [review]

**Pull request merged**